### PR TITLE
front: update btn map in scenario page and convert rem to px in css map file

### PR DESCRIPTION
**Pull request by theocrsb**
closes https://github.com/osrd-project/osrd-confidential/issues/670

buttons list mockup: https://www.sketch.com/s/2fb064cd-c483-46ac-bd30-72705fd63e75/p/9559F6FC-69EE-4DB8-BE1A-77F60F10C97F/canvas#Inspect

button mockup: https://www.sketch.com/s/7b92c9be-3fa6-4af7-b8c6-66b63fec08fa/p/AB9D17BF-0EBF-4C30-A75D-0D6BB5C0BA68/canvas#Inspect

### CODE REVIEW

File: front/src/styles/scss/common/map/_map.scss
Comment (anisometropie) [inline front/src/styles/scss/common/map/_map.scss]: where does 48px come from?
Screenshot_20241121_165841

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/styles/scss/common/map/_map.scss
Comment (theocrsb) [reply front/src/styles/scss/common/map/_map.scss]: I used the mockup `Graphic update for OSRD#2`
image

File: front/src/common/Map/Buttons/MapButtons.tsx
Comment (clarani) [inline front/src/common/Map/Buttons/MapButtons.tsx]: Set it to false by default, this way you can remove the props in the components which do not need the new buttons
```suggestion
  isNewButtons = false,
```

File: front/src/applications/editor/Editor.tsx
Comment (clarani) [inline front/src/applications/editor/Editor.tsx]: You can remove this props if it's set by default to false
```suggestion
```

**Review by clarani — COMMENTED** [review]
As you are re-writting these buttons, I think it would be better to do a bit of refacto :)
All these buttons are almost the same, the only things that change are the function onClick, the text in the span + its translation, the icon and the data-test-id.

You can create a generic component `MapButton` which will take only these 4-5 props, and in MapButtons, render this new component with the correct props if we are using the newButtons.

If you do this, you won't need to modify all the existing Buttons

File: front/src/common/Map/Buttons/MapButtons.tsx
Comment (theocrsb) [reply front/src/common/Map/Buttons/MapButtons.tsx]: Done

File: front/src/applications/editor/Editor.tsx
Comment (theocrsb) [reply front/src/applications/editor/Editor.tsx]: Done

**Comment by theocrsb** [comment]
> As you are re-writting these buttons, I think it would be better to do a bit of refacto :) All these buttons are almost the same, the only things that change are the function onClick, the text in the span + its translation, the icon and the data-test-id.
> 
> You can create a generic component `MapButton` which will take only these 4-5 props, and in MapButtons, render this new component with the correct props if we are using the newButtons.
> 
> If you do this, you won't need to modify all the existing Buttons

You're right, I just proposed a refacto in my fixups :)

File: front/src/styles/scss/common/map/_map.scss
Comment (anisometropie) [reply front/src/styles/scss/common/map/_map.scss]: thanks. perhaps link it in the description so we know what you used as reference

Comment (theocrsb) [reply front/src/styles/scss/common/map/_map.scss]: Sure

**Comment by anisometropie** [comment]
I think all the icons aren’t centered 
Screenshot_20241128_102413

**Comment by theocrsb** [comment]
> I think all the icons aren’t centered Screenshot_20241128_102413

Thanks, it's done

File: front/src/styles/scss/common/map/_map.scss
Comment (clarani) [inline front/src/styles/scss/common/map/_map.scss]: Can you move `.osrd-btn-map` inside osrd-btn-map-container ?
Also, maybe we don't need the prefix `osrd` ? You can replace it by the prefix `new` if you need

**Review by clarani — COMMENTED** [review]
There is a bug in the old buttons :/ Look at the compass button
image

File: front/src/styles/scss/common/map/_map.scss
Comment (theocrsb) [reply front/src/styles/scss/common/map/_map.scss]: Done

**Comment by theocrsb** [comment]
> There is a bug in the old buttons :/ Look at the compass button image

Thanks, I just fixed

**Review by anisometropie — APPROVED** [review]

**Comment by theocrsb** [comment]
It doesn't yet have “hover” and “active” behavior. 
I've just checked with @thibautsailly , the mockup are in progress.

File: front/src/common/Map/Buttons/MapButton.tsx
Comment (clarani) [inline front/src/common/Map/Buttons/MapButton.tsx]: nit: it's the prop of 1 button
```suggestion
  isNewButton: boolean;
```

File: front/src/styles/scss/common/map/_map.scss
Comment (clarani) [inline front/src/styles/scss/common/map/_map.scss]: You can move this class in `new-btn-map-container`

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark: Only left 2 small comments, you can close them once you have fixed them and merge without a new approval :+1: :)

File: front/src/common/Map/Buttons/MapButton.tsx
Comment (theocrsb) [reply front/src/common/Map/Buttons/MapButton.tsx]: Done

File: front/src/styles/scss/common/map/_map.scss
Comment (theocrsb) [reply front/src/styles/scss/common/map/_map.scss]: Done

**Pull request merged**
